frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -136,35 +143,93 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
-      ...(subscription?.status?.installedCSV
-        ? [
-            k8sKill(
-              ClusterServiceVersionModel,
-              {
-                metadata: {
-                  name: subscription.status.installedCSV,
-                  namespace: subscription.metadata.namespace,
+    const uninstallOperator = () => {
+      const operatorUninstallPromises = [
+        k8sKill(SubscriptionModel, subscription, {}, deleteOptions),
+        ...(subscription?.status?.installedCSV
+          ? [
+              k8sKill(
+                ClusterServiceVersionModel,
+                {
+                  metadata: {
+                    name: subscription.status.installedCSV,
+                    namespace: subscription.metadata.namespace,
+                  },
                 },
-              },
-              {},
-              deleteOptions,
-            ),
-          ]
-        : []),
-      ...(removePlugins
-        ? [k8sPatch(ConsoleOperatorConfigModel, consoleOperatorConfig, [patch])]
-        : []),
-    ];
-    const operandDeletionPromises = deleteOperands
-      ? operands.map((operand: K8sResourceCommon) => {
-          const model = modelFor(referenceFor(operand));
-          return k8sKill(model, operand, {}, deleteOptions);
+                {},
+                deleteOptions,
+              ),
+            ]
+          : []),
+        ...(removePlugins
+          ? [k8sPatch(ConsoleOperatorConfigModel, consoleOperatorConfig, [patch])]
+          : []),
+      ];
+
+      handleOperatorUninstallPromise(Promise.all(operatorUninstallPromises))
+        .then(() => {
+          setOperatorUninstallFinished(true);
         })
-      : [];
-    if (operandDeletionPromises.length > 0) {
+        .catch(() => {
+          setOperatorUninstallFinished(true);
+        });
+    };
+
+    const setOperandErrorsTo = (
+      operators: K8sResourceCommon[],
+      errorMsg: string,
+    ): OperandError[] => {
+      return operators.reduce((acc: OperandError[], curr, i) => {
+        return acc.concat({
+          operand: operands[i],
+          errorMessage: errorMsg,
+        });
+      }, []);
+    };
+
+    const verifyDeletionOfOperands = () => {
+      const url = `${window.SERVER_FLAGS.basePath}api/list-operands?name=${subscriptionName}&namespace=${subscriptionNamespace}`;
+      let operandVerificationErrors: OperandError[] = [];
+      let intervalID: Timeout;
+
+      const finishVerification = () => {
+        clearInterval(intervalID);
+        setOperandsDeleteInProgress(false);
+        setOperandsDeleteFinished(true);
+        if (operandVerificationErrors.length === 0) {
+          uninstallOperator();
+        } else {
+          // show operand deletion verification errors on results page
+          setOperandDeletionErrors(operandVerificationErrors);
+          setOperatorUninstallFinished(true);
+        }
+      };
+
+      intervalID = setInterval(() => {
```

Comment:
It doesn't look to me like there is anything that guarantees the interval is cleared (for instance, if the user closes the modal). In which case we'll continue polling this endpoint every 2 seconds until the user closes the tab.

We might need to call `setInterval` inside a `useEffect` hook where we can ensure the interval is cleared. You could use `operandsDeleteInProgress` in the `useEffect` dependency list to know when to poll.

Reply:
Great catch @spadgett!  Last commit has blocking modal from being closed by clicking elsewhere and clicking Cancel button will also clear the interval.